ArduCopter/afs_copter.cpp

```diff
@@ -17,20 +17,26 @@ AP_AdvancedFailsafe_Copter::AP_AdvancedFailsafe_Copter(AP_Mission &_mission, AP_
  */
 void AP_AdvancedFailsafe_Copter::terminate_vehicle(void)
 {
-    // stop motors
-    copter.motors->set_desired_spool_state(AP_Motors::DESIRED_SHUT_DOWN);
-    copter.motors->output();
+    if (_terminate_action == LAND) {
+        if (copter.control_mode != LAND) {
```

Comment:
this IF is not needed, the set_mode function will just return if the mode matches. simply always call it

Reply:
Didn't know what copter did with set_mode, I guess I should have peaked at it to see what happened if you repeatedly called the same mode.